IllegalArgumentException on completionItem/resolve of package declaration

Started happening on JDT-LS 1.24.0. Likely due to https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2646 . It doesn't seem too harmful as not much happens for package declarations on resolve.

It seems like the root of the issue is that package declaration proposals return a type declaration signature (eg. `java.util` ) that isn't a real signature, but we pass it to the signature processing utility methods, which ultimately fail.

**Test.java**
```
package org.example;

public class Test {
    public void test() {
        java|
    }
}
```

```
[Info  - 10:31:58] Oct. 24, 2023, 10:31:58 a.m. >> document/resolveCompletionItem
Oct. 24, 2023 10:31:58 A.M. org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: java.lang.IllegalArgumentException: "java" at 0
java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: "java" at 0
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.uniAcceptNow(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.uniAcceptStage(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.thenAccept(Unknown Source)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:279)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.jdt.ls.core.internal.ParentProcessWatcher.lambda$1(ParentProcessWatcher.java:144)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: "java" at 0
	at org.eclipse.jdt.core.Signature.newIllegalArgumentException(Signature.java:3017)
	at org.eclipse.jdt.core.Signature.appendTypeSignature(Signature.java:972)
	at org.eclipse.jdt.core.Signature.appendTypeSignature(Signature.java:875)
	at org.eclipse.jdt.core.Signature.toCharArray(Signature.java:2753)
	at org.eclipse.jdt.core.Signature.toString(Signature.java:2959)
	at org.eclipse.jdt.internal.corext.template.java.SignatureUtil.stripSignatureToFQN(SignatureUtil.java:206)
	at org.eclipse.jdt.ls.core.internal.handlers.CompletionResolveHandler.resolve(CompletionResolveHandler.java:210)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.resolveCompletionItem(JDTLanguageServer.java:652)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$recursiveFindRpcMethods$0(GenericEndpoint.java:65)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	... 10 more
```

--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionResolveHandler.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionResolveHandler.java
@@ -128,7 +128,7 @@ public class CompletionResolveHandler {
 					if (manager.getClientPreferences().isCompletionResolveDocumentSupport()) {
 						param.setDocumentation(SnippetUtils.beautifyDocument(content));
 					}
-	
+
 					if (manager.getPreferences().isCompletionLazyResolveTextEditEnabled()) {
 						SnippetCompletionProposal.setTextEdit(ctx, unit, param, content);
 					}
@@ -156,7 +156,7 @@ public class CompletionResolveHandler {
 						PostfixTemplateEngine.setAdditionalTextEdit(param, unit, postfixContext, range, template);
 					}
 				}
-	
+
 				param.setData(null);
 			} catch (JavaModelException e) {
 				JavaLanguageServerPlugin.logException(e.getMessage(), e);
@@ -207,7 +207,12 @@ public class CompletionResolveHandler {
 		} else {
 			char[] declarationSignature = proposal.getDeclarationSignature();
 			if (declarationSignature != null) {
-				String typeName = stripSignatureToFQN(String.valueOf(declarationSignature));
+				String typeName = String.valueOf(declarationSignature);
+				try {
+					typeName = stripSignatureToFQN(typeName);
+				} catch (IllegalArgumentException e) {
+					// simple names / dot-based package names
+				}
 				try {
 					IType type = unit.getJavaProject().findType(typeName);
 					if (type != null && proposal.getName() != null) {

